src/property.ts

```diff
@@ -953,7 +930,37 @@ function Property$ensureInited(property: Property, obj: Entity): void {
 
 		// Do not initialize calculated properties. Calculated properties should be initialized using a property get rule.
 		if (!property.isCalculated) {
-			Property$init(property, obj, Property$getInitialValue(property, obj));
+			let initData = null;
+			let initValue = null;
+
+			// Constant
+			if (property.isConstant)
+				initValue = typeof property.constant === "function" ? (property.constant = property.constant()) : property.constant;
+			else if (property.initializer) {
+				initValue = property.initializer.call(obj);
+				if (isEntityType(property.propertyType)) {
+					initData = initValue;
+					if (property.isList)
+						initValue = [];
+					else
+						initValue = new property.propertyType();
+				}
+			}
+			else
+				initValue = property.defaultValue;
+
+			if (Array.isArray(initValue)) {
+				initValue = ObservableArray.ensureObservable(initValue as any[]);
+
+				// Override the default toString on arrays so that we get a comma-delimited list
+				// TODO: Implement toString on observable list?
+				// val.toString = Property$_arrayToString.bind(val);
+			}
+
+			Property$init(property, obj, initValue);
+
+			if (initData)
+				obj.update(property.name, initData);
```

Comment:
When using the `init` option, after setting the property in the model, call `update` to initialize the object (ex: if the `init` function returned JSON).